src/ucp/wireup/select.c

```diff
@@ -1426,29 +1426,101 @@ ucp_wireup_is_md_map_count_valid(ucp_context_h context, ucp_md_map_t md_map)
            (ucs_popcount(md_map) < UCP_MAX_OP_MDS);
 }
 
+static double ucp_wireup_get_lane_bw(const ucp_wireup_select_info_t *sinfo,
+                                     ucp_worker_h worker,
+                                     const ucp_address_entry_t *address)
+{
+    ucp_context_h context = worker->context;
+    const uct_iface_attr_t *iface_attr;
+    double bw_local, bw_remote;
+
+    iface_attr = ucp_worker_iface_get_attr(worker, sinfo->rsc_index);
+    bw_local   = ucp_tl_iface_bandwidth(context, &iface_attr->bandwidth);
+    bw_remote  = address[sinfo->addr_index].iface_attr.bandwidth;
+
+    return ucs_min(bw_local, bw_remote);
+}
+
+static unsigned
+ucp_wireup_add_fast_lanes(const ucp_wireup_select_params_t *select_params,
+                          const ucp_wireup_select_info_t *sinfo_array,
+                          unsigned num_sinfo, ucp_worker_h worker,
+                          ucp_lane_type_t lane_type,
+                          ucp_wireup_select_context_t *select_ctx)
+{
+    unsigned num_lanes     = 0;
+    double max_bw          = 0;
+    ucp_context_h context  = worker->context;
+    const double max_ratio = 1. / context->config.ext.multi_lane_max_ratio;
+    ucs_status_t status;
+    int show_error;
+    unsigned sinfo_index;
+    double lane_bw;
+    const ucp_wireup_select_info_t *sinfo;
+
+    if (num_sinfo == 0) {
+        return 0;
+    }
+
+    /* Iterate over all elements and calculate max BW */
+    for (sinfo_index = 0; sinfo_index < num_sinfo; ++sinfo_index) {
+        lane_bw = ucp_wireup_get_lane_bw(&sinfo_array[sinfo_index], worker,
+                                         select_params->address->address_list);
+        max_bw  = ucs_max(lane_bw, max_bw);
+    }
+
+    ucs_assertv(max_bw > 0, "max_bw has illegal value: %f", max_bw);
```

Comment:
Which problem does the assert reveal?
Maybe it is better replace line 1480 by `if (lane_bw < (max_bw * max_ratio)) {`, and get rid of assert, if the only purpose of the assert is to avoid division by zero.